.github/workflows/code-quality.yml

```diff
@@ -3,6 +3,10 @@ on: [push, pull_request]
 jobs:
   code-quality:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Latest released and oldest supported.
+        node: [17, 12]
```

Comment:
to cover ranges:
```
node: [17, 16, 14, 12]
```

Reply:
To my mind the chances of the code working in both Node v12 and v17 but not in v14 or v16 is vanishingly small to the point that it feels like a waste of energy.

Reply:
> to cover ranges:
> 
> ```
> node: [17, 16, 14, 12]
> ```
> 
> Why cutting corners here?

Probably best to keep 17 as you've mentioned, but I removed 12 because it's coming out of LTS.

Reply:
OK, I think it's fine since now targeting to **es2019** which has 100% features compatible through out **v12** to **v18**.